File: swastricare-mobile-swift/Helpers/UserFriendlyError.swift
import Foundation

enum UserFriendlyError {
    static func message(from error: Error) -> String {
        let desc = error.localizedDescription.lowercased()

        if desc.contains("network") || desc.contains("connection") || desc.contains("offline") || desc.contains("internet") {
            return "No internet connection. Please check your network and try again."
        }
        if desc.contains("timeout") || desc.contains("timed out") {
            return "Request timed out. Please try again."
        }
        if desc.contains("unauthorized") || desc.contains("not authenticated") || desc.contains("401") {
            return "Your session has expired. Please sign in again."
        }
        if desc.contains("permission") || desc.contains("denied") || desc.contains("403") {
            return "You don't have permission for this action."
        }
        if desc.contains("not found") || desc.contains("404") {
            return "The requested data could not be found."
        }
        if desc.contains("rate limit") || desc.contains("too many") {
            return "Too many requests. Please wait a moment and try again."
        }
        if desc.contains("server") || desc.contains("500") || desc.contains("503") {
            return "Our servers are temporarily unavailable. Please try again later."
        }
        if desc.contains("cancelled") || desc.contains("canceled") {
            return "The operation was cancelled."
        }
        if desc.contains("storage") || desc.contains("disk") || desc.contains("space") {
            return "Not enough storage space. Please free up some space and try again."
        }

        // Default friendly message
        return "Something went wrong. Please try again."
    }
}

File: android/app/src/main/kotlin/com/swastricare/health/core/UserFriendlyError.kt
package com.swastricare.health.core

object UserFriendlyError {
    fun from(e: Exception): String {
        val msg = e.message?.lowercase() ?: return "Something went wrong. Please try again."
        return when {
            "network" in msg || "connection" in msg || "internet" in msg || "unable to resolve" in msg ->
                "No internet connection. Please check your network and try again."
            "timeout" in msg || "timed out" in msg ->
                "Request timed out. Please try again."
            "unauthorized" in msg || "not authenticated" in msg || "401" in msg ->
                "Your session has expired. Please sign in again."
            "permission" in msg || "denied" in msg || "403" in msg ->
                "You don't have permission for this action."
            "not found" in msg || "404" in msg ->
                "The requested data could not be found."
            "rate limit" in msg || "too many" in msg ->
                "Too many requests. Please wait a moment and try again."
            "server" in msg || "500" in msg || "503" in msg ->
                "Our servers are temporarily unavailable. Please try again later."
            else -> "Something went wrong. Please try again."
        }
    }
}
